lib/result/result.ts

```diff
@@ -141,6 +141,14 @@ export function Err<T, E>(err: E): ResErr<T, E> {
   };
 }
 
+export function ResultOf<T, E = Error>(fallible: () => T): Result<T, E> {
```

Comment:
It's worth noting that any value can be thrown and it's impossible to know in advance if it's an error type. This is why TypeScript doesn't set a type in catch blocks either. Based on this, I suggest setting the default value to unknown.

Reply:
I do believe that an error is best represented with the `Error` class.  But, I understand this is more of an opinion so i will change that  to `unknown`